ecommerce/Supports/Shop.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * Settings support class.
+ *
+ * @package Tutor\Ecommerce
+ * @author Themeum
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\Ecommerce\Supports;
+
+use Tutor\Ecommerce\OptionKeys;
+use Tutor\Ecommerce\Supports\Settings;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+final class Shop {
+	public static function as_negative( string $price ) {
+		return '−&thinsp;' . $price;
+	}
+
+	public static function get_tax_settings() {
+		$tax_settings = tutor_utils()->get_option( 'ecommerce_tax' );
+
+		if ( ! empty( $tax_settings ) && is_string( $tax_settings ) ) {
+			$tax_settings = json_decode( $tax_settings );
+		}
+
+		return $tax_settings;
+	}
+
+	/**
+	 * add the object properties with currency format.
+	 * will search a key by the provided keys array and if exists and the value is numeric then
+	 * create a new key with the _with_currency suffix and format the numeric value with currency symbol.
+	 *
+	 * @param object $data
+	 * @param array  $keys
+	 *
+	 * @return object
+	 */
+	public static function format_with_currency( object $data, array $keys = array() ) {
+		if ( empty( $keys ) ) {
+			$keys = array_keys( (array) $data );
+		}
+
+		foreach ( $keys as $key ) {
+			if ( ! isset( $data->$key ) || ! is_numeric( $data->$key ) ) {
+				continue;
+			}
+
+			$new_key        = $key . '_with_currency';
+			$data->$new_key = tutor_get_formatted_price( $data->$key );
+		}
+
+		return $data;
+	}
+
+	public static function calculate_taxable_amount( $price, $tax_rate ) {
+		$price    = $price ?? 0;
```

Comment:
I think Instead null operator we should use the ternary operator
```
$price    = $price ?? 0;
```
`$price = $price ? $price : 0`
For the null operator if only the value $price is null then it will set 0 if price is not null like empty string then $price will be set as empty string